File: 7-kyu/find-the-nth-digit-of-a-number/swift/solution.swift
import Foundation

func findDigit(_ num:Int, _ nth: Int) -> Int {
  guard nth > 0 else { return -1 }
  guard nth <= String(num).count else { return 0 }
  let numArr = abs(num).description.compactMap { Int(String($0)) }
  return numArr[numArr.count - nth]
}

File: 7-kyu/find-the-nth-digit-of-a-number/c++/solution.cpp
#include <cmath>
#include <string>

int findDigit(int num, int nth)
{
    if (nth < 1) return -1;
    std::string n = std::to_string(std::abs(num));
    int index = n.length() - nth;
    if (index < 0) return 0;
    return n[index] - '0';
}